internal/core/software_renderer.rs

```diff
@@ -2495,9 +2542,100 @@ impl<'a, T: ProcessScene> crate::item_rendering::ItemRenderer for SceneBuilder<'
         }
     }
 
-    #[cfg(feature = "std")]
-    fn draw_path(&mut self, _path: Pin<&crate::items::Path>, _: &ItemRc, _size: LogicalSize) {
-        // TODO
+    #[cfg(feature = "path")]
+    fn draw_path(&mut self, path: Pin<&crate::items::Path>, item: &ItemRc, size: LogicalSize) {
+        use zeno::PathBuilder;
+        let geom = LogicalRect::from(size);
+
+        let clipped = match geom.intersection(&self.current_state.clip) {
+            Some(geom) => geom,
+            None => return,
+        };
+
+        let geometry = (clipped.translate(self.current_state.offset.to_vector()).cast()
+            * self.scale_factor)
+            .round()
+            .cast()
+            .transformed(self.rotation);
+
+        let path_props = path.as_ref();
+        let mut zeno_pb: Vec<zeno::Command> = Vec::new();
+        let (logical_offset, path_events2) = path.fitted_path_events(item).unwrap();
+
+        for event in path_events2.iter() {
+            match event {
+                Event::Begin { at } => {
+                    zeno_pb.move_to([at.x, at.y]);
+                }
+                Event::Line { from: _, to } => {
+                    zeno_pb.line_to([to.x, to.y]);
+                }
+                Event::Quadratic { from: _, ctrl, to } => {
+                    zeno_pb.quad_to(
+                        [ctrl.x, ctrl.y],
+                        [to.x, to.y]
+                    );
+                }
+                Event::Cubic { from: _, ctrl1, ctrl2, to } => {
+                    zeno_pb.curve_to(
+                        [ctrl1.x, ctrl1.y],
+                        [ctrl2.x, ctrl2.y],
+                        [to.x, to.y],
+                    );
+                }
+                Event::End { last: _, first: _, close    } => {
+                    if close {
+                        zeno_pb.close();
+                    }
+                }
+            }
+        }
+
+        let transform = Some(zeno::Transform::translation(logical_offset.x, logical_offset.y));
+
+        let fill_mask = if !path_props.fill().is_transparent() {
+            let mut mask = Vec::new();
+            mask.resize((geometry.size.width * geometry.size.height) as usize, 0u8);
```

Comment:
That seems big. 
Internally, zeno renders row by row as well, I wonder if there is a way to do the actual path rendering in draw_zeno_path_line instead. (assuming that the storage needed in the rasterizer is less)
But anyway, there doesn't seem to be API in zeno to do that.

Reply:
yes, it doesn't seem to be able to render it line by line. at least the command saves two u8 buffers, which is less if it was a rgba buffer. On some occasions, there may be big blank areas, like when full sizing on flex layouts. Zeno provides a way to make the buffer on the needed size, but it would need to limit this somehow (prevent huge buffers with overflow lines) and have a reference point to translate the buffer to the correct location